File: 01.backend/src/main/java/com/fallapp/dto/PaginatedResponse.java
package com.fallapp.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Wrapper para respuestas paginadas
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PaginatedResponse<T> {
    private List<T> contenido;
    private Integer paginaActual;
    private Integer elementosPorPagina;
    private Long totalElementos;
    private Integer totalPaginas;
    private Boolean esUltimaPagina;
}

File: 03.mobile/app/src/main/java/com/fallapp/core/network/PaginatedResponse.kt
package com.fallapp.core.network

import kotlinx.serialization.Serializable

/**
 * DTO genérico para respuestas paginadas de la API.
 * 
 * Estructura devuelta por endpoints como:
 * - GET /api/fallas
 * - GET /api/eventos
 * - GET /api/ninots
 * 
 * @param T tipo del contenido paginado
 */
@Serializable
data class PaginatedResponse<T>(
    val contenido: List<T>,
    val paginaActual: Int,
    val elementosPorPagina: Int,
    val totalElementos: Long,
    val totalPaginas: Int,
    val esUltimaPagina: Boolean
)
